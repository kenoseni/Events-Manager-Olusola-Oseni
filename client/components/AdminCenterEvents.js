import React, {Component} from 'react';
import AdminCenterEventsTable from './AdminCenterEventsTable';

class AdminCenterEvents extends Component {
  constructor(props) {
    super(props);
  }

  render () {
    const {events} = this.props
    return (
      <div className="col-xs-12 col-md-8" id="etable">
        <div className="card w-70">
          <div className="card-body">
            <div className="table-wrapper-2">
              {/*<!--Table-->*/}
              <table className="table table-responsive">
                <thead className="mdb-color lighten-4">
                  <tr>
                    <th>#</th>
                    <th className="th-lg">Event Type</th>
                    <th className="th-lg">Start Date</th>
                    <th className="th-lg">End Date</th>
                    <th className="th-lg">Time</th>
                  </tr>
                </thead>
                <tbody>
                {(events === undefined ? null: events.map((event, i) => <AdminCenterEventsTable i={i+1} key={event.id} event={event}  />))}
                </tbody>
              </table>
              {/*<!--Table-->*/}
            </div>
          </div>
        </div>
      </div>
    )
  }
}
export default AdminCenterEvents;
