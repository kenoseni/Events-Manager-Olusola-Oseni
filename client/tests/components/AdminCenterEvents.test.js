import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AdminCenterEvents from '../../components/AdminCenterEvents';

configure({ adapter: new Adapter() });

let props = {
  events: []
}

const wrapper = shallow(<AdminCenterEvents
  {...props}
/>)

describe('<AdminCenterDetails />', () => {
  it('should render a <AdminCenterEvents /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('it should  not return null if events is undefined', () => {
    wrapper.setProps({
      ...wrapper.props,
      events: undefined
      
    })
    expect(wrapper.find('tbody').at(0).text()).toBeNull
  });
  it('it should  render AdminCenterEventsTable if events is not undefined', () => {
    wrapper.setProps({
      ...wrapper.props,
      events: ['quiz', 'seminar']
      
    })
    expect(wrapper.find('AdminCenterEventsTable').exists()).toBeTruthy()
  });
})
