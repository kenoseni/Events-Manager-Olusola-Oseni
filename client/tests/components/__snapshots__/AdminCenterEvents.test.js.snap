// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminCenterDetails /> should render a <AdminCenterEvents /> component 1`] = `
<div
  className="col-xs-12 col-md-8"
  id="etable"
>
  <div
    className="card w-70"
  >
    <div
      className="card-body"
    >
      <div
        className="table-wrapper-2"
      >
        <table
          className="table table-responsive"
        >
          <thead
            className="mdb-color lighten-4"
          >
            <tr>
              <th>
                #
              </th>
              <th
                className="th-lg"
              >
                Event Type
              </th>
              <th
                className="th-lg"
              >
                Start Date
              </th>
              <th
                className="th-lg"
              >
                End Date
              </th>
              <th
                className="th-lg"
              >
                Time
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
